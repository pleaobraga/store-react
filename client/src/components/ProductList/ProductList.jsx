import React from 'react'
import PropTypes from 'prop-types'
import List from '../List'
import Product from '../Product'

import './ProductList.scss'

export const ProductList = ({ listItems }) => {
  return (
    <List
      className="product-list"
      header={
        <>
          <p key="Name" className="header__name">
            Name
          </p>
          <p key="Name" className="header__quantity">
            Quantity
          </p>
          <p key="Name" className="header__quantity abbr">
            Qty
          </p>
          <p key="Name" className="header__price">
            Price
          </p>
        </>
      }
      listItems={
        <>
          {listItems.map(item => (
            <Product
              key={item.id}
              name={item.name}
              quantity={item.quantity}
              price={item.price}
              currency={item.currency}
            />
          ))}
        </>
      }
    />
  )
}

ProductList.defaultProps = {
  listItems: []
}

ProductList.propTypes = {
  listItems: PropTypes.arrayOf(PropTypes.shape(Product.propTypes))
}

export default React.memo(ProductList)
