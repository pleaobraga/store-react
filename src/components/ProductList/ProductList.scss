@import '../../styles/colors';

.product-list {
  .list__header {
    .header {
      &__name {
        width: 60%;
      }

      &__quantity {
        text-align: end;
        width: 15%;
      }

      &__price {
        text-align: end;
        width: 35%;
      }
    }
  }
}
