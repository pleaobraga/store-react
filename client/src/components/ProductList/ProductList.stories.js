import React from 'react'
import { storiesOf } from '@storybook/react'
import { BrowserRouter as Router } from 'react-router-dom'
import { ProductList } from './ProductList'

const listItems = [
  {
    id: '0',
    name: 'Product 1',
    quantity: 1,
    price: '10,00',
    currency: 'R$'
  },
  {
    id: '1',
    name: 'Product 2',
    quantity: 2,
    price: '20,00',
    currency: 'R$'
  },
  {
    id: '2',
    name: 'Product 3',
    quantity: 3,
    price: '30,00',
    currency: 'R$'
  },
  {
    id: '3',
    name: 'Product 4',
    quantity: 4,
    price: '40,00',
    currency: 'R$'
  }
]

export default storiesOf('Components | ProductList', module).add(
  'default',
  () => (
    <Router>
      <ProductList listItems={listItems} />
    </Router>
  ),
  { info: { inline: true, header: false, propTables: [ProductList] } }
)
