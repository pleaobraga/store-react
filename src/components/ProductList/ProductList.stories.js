import React from 'react'
import { storiesOf } from '@storybook/react'
import { BrowserRouter as Router } from 'react-router-dom'
import ProductList from './ProductList'

const listItems = [
  {
    id: '0',
    name: 'Name',
    quantity: '1',
    price: '10,00',
    currency: 'R$'
  }
]

export default storiesOf('Components | ProductList', module).add(
  'default',
  () => (
    <Router>
      <ProductList listItems={listItems} />
    </Router>
  ),
  { info: { inline: true, header: false } }
)
