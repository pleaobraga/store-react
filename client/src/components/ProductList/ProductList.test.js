import React from 'react'
import { shallow } from 'enzyme'
import ProductList from './ProductList'

describe('ProductList', () => {
  const listItem = shallow(<ProductList />)

  it('render proprely', () => {
    expect(listItem).toMatchSnapshot()
  })
})
