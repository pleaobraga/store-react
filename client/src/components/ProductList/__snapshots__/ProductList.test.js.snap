// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList render empty List 1`] = `
<Memo(List)
  className="product-list"
  header={
    <React.Fragment>
      <p
        className="header__name"
      >
        Name
      </p>
      <p
        className="header__quantity"
      >
        Quantity
      </p>
      <p
        className="header__quantity abbr"
      >
        Qty
      </p>
      <p
        className="header__price"
      >
        Price
      </p>
    </React.Fragment>
  }
  listItems={
    <p
      className="empty-message"
    >
      Empty Product List
    </p>
  }
  title="Products"
/>
`;

exports[`ProductList render properly 1`] = `
<Memo(List)
  className="product-list"
  header={
    <React.Fragment>
      <p
        className="header__name"
      >
        Name
      </p>
      <p
        className="header__quantity"
      >
        Quantity
      </p>
      <p
        className="header__quantity abbr"
      >
        Qty
      </p>
      <p
        className="header__price"
      >
        Price
      </p>
    </React.Fragment>
  }
  listItems={
    <React.Fragment>
      <UNDEFINED
        currency="R$"
        id="0"
        name="Product 1"
        price="10,00"
        quantity={1}
      />
      <UNDEFINED
        currency="R$"
        id="1"
        name="Product 2"
        price="20,00"
        quantity={2}
      />
      <UNDEFINED
        currency="R$"
        id="2"
        name="Product 3"
        price="30,00"
        quantity={3}
      />
      <UNDEFINED
        currency="R$"
        id="3"
        name="Product 4"
        price="40,00"
        quantity={4}
      />
    </React.Fragment>
  }
  title="Products"
/>
`;
